common/web/gesture-recognizer/src/gestureRecognizer.ts

```diff
@@ -0,0 +1,64 @@
+/// <reference path="gestureRecognizerConfiguration.ts" />
+/// <reference path="includes/events.ts" />
+
+namespace com.keyman.osk {
+  type Mutable<Type> = {
+    -readonly [Property in keyof Type]: Type[Property];
+  };
+
+  export enum TrackedInputState {
+    START = "start",
+    END = "end",
+    MOVE = "move",
+    CANCEL = "cancel"
+  };
+
+  export class GestureRecognizer extends EventEmitter {
+    public static readonly TRACKED_INPUT_UPDATE_EVENT_NAME = "trackedInputUpdate";
+
+    protected readonly config: GestureRecognizerConfiguration;
+
+    private readonly mouseEngine?: MouseEventEngine;
+    private readonly touchEngine?: TouchEventEngine;
+
+    protected static preprocessConfig(config: GestureRecognizerConfiguration): GestureRecognizerConfiguration {
+      // Allows configuration pre-processing during this method.
+      let processingConfig: Mutable<GestureRecognizerConfiguration> = Object.assign({}, config);
```

Comment:
This should be consistent with using object spread operator, yeah? We should use ES6 syntax where we can:

```suggestion
      let processingConfig: Mutable<GestureRecognizerConfiguration> = {...config};
```

Reply:
Maybe one day I'll remember I can do this without prompting.